apps/buildog/src/components/user-auth-form.tsx

```diff
@@ -15,10 +16,25 @@ export function UserAuthForm({ className, ...props }: UserAuthFormProps) {
   const router = useRouter();
   async function onSubmit(event: React.SyntheticEvent) {
     event.preventDefault();
-    /*
-      handle login process
-    */
-    router.push("/blog/");
+
+    const email = event.target[0].value;
+    const password = event.target[1].value;
+
+    try {
+      Auth.authenticate({
+        email: email,
+        password: password,
+      });
+
+      router.push("/blog/");
+    } catch (error) {
+      console.log(error);
+      /*
+        Handle login error
+      */
+    }
```

Comment:
We don't need try-catch here, in Auth.authentication already have on. I got you, you struggle how to get information that Auth.authentication will work or not. Is it possible to add return boolean on Auth.authenticate function?

```TypeScript
  async authenticate(credentials: Credentials): Promise<
    auth: boolean;
    error?: { error?: string; error_description: string };
    > {
    try {
      // ...code
      return {auth: true};
    } catch (error: any) {
      this.emit("authentication_failed");

      // Check if the error has the expected structure
      if (error.response && typeof error.response.code === "string") {
        return { auth: false, error: error.response.data };
      } else {
        // Handle unexpected error formats
        return {
          auth: false,
          error: { error_description: "An unknown error occurred" },
        };
      }
    }
  }
```

then you can use await and understand what returns just like that;

```TypeScript
  async function onSubmit(event: React.SyntheticEvent) {
    event.preventDefault();

    const email = event.target[0].value;
    const password = event.target[1].value;

    setIsLoading(true);
    const authSuccess = await Auth.authenticate({
      email: email,
      password: password,
    });

    if (authSuccess.auth) router.push("/blog/");
    else {
       // show error to user.
       console.log(authSuccess.error);
    }
  }
```